apps/dash-stitching/app.py

```diff
@@ -0,0 +1,495 @@
+        Input("stitched-res", "children"),
+    ],
+)
+def modify_result(contrast, brightness, image_string):
+    if image_string is None:
+        raise PreventUpdate
+    img = np.asarray(image_string_to_PILImage(image_string))
+    img = contrast_adjust(img, contrast)
+    img = brightness_adjust(img, brightness)
+    return array_to_data_url(img)
+
+
+@app.callback(
+    Output("stitched-res", "children"),
+    [Input("button-stitch", "n_clicks")],
+    [
+        State("nrows-stitch", "value"),
+        State("ncolumns-stitch", "value"),
+        State("overlap-stitch", "value"),
+        State("table-stitch", "data"),
+        State("sh_x", "children"),
+        State("do-blending-stitch", "values"),
+    ],
+)
+def modify_content(n_cl, n_rows, n_cols, overlap, estimate, image_string, vals):
+    blending = 0
+    if vals is not None:
+        blending = 1 in vals
+    if image_string is None:
+        raise PreventUpdate
+    tiles = untile_images(image_string, n_rows, n_cols)
+    if estimate is not None and len(estimate) > 0:
+
+        overlap_dict = _sort_props_lines(
+            estimate, tiles.shape[2], tiles.shape[3], n_cols
+        )
+    else:
+
+        overlap_dict = None
+    canvas = register_tiles(
+        tiles,
+        n_rows,
+        n_cols,
+        overlap_global=overlap,
+        overlap_local=overlap_dict,
+        pad=np.max(tiles.shape[2:]),
+        blending=blending,
+    )
+    return array_to_data_url(canvas)
+
+
+@app.callback(Output("canvas-stitch", "image_content"), [Input("sh_x", "children")])
+def update_canvas_image(im):
+    return im
+
+
+@app.callback(Output("upload-stitch", "contents"), [Input("demo", "n_clicks")])
+def reset_contents(n_clicks):
+    if n_clicks:
+        return None
+
+
+@app.callback(
+    Output("sh_x", "children"),
+    [
+        Input("upload-stitch", "contents"),
+        Input("upload-stitch", "filename"),
+        Input("demo", "n_clicks"),
+        Input("downsample", "value"),
+    ],
+    [State("nrows-stitch", "value"), State("ncolumns-stitch", "value")],
+)
+def upload_content(
+    list_image_string, list_filenames, click, downsample, n_rows, n_cols
+):
+
+    downsample = int(downsample)
+    if list_image_string is not None:
+        order = np.argsort(list_filenames)
+        image_list = [
+            np.asarray(image_string_to_PILImage(list_image_string[i])) for i in order
+        ]
+        if downsample > 1:
+            ratio = 1.0 / downsample
+            multichannel = image_list[0].ndim > 2
+            image_list = [
+                transform.rescale(
+                    image, ratio, multichannel=multichannel, preserve_range=True
+                ).astype(np.uint8)
+                for image in image_list
+            ]
+        res = tile_images(image_list, n_rows, n_cols)
+        return array_to_data_url(res)
+    elif click:
+        res = demo_data()
+        tmp = array_to_data_url(res)
+        return tmp
+    else:
```

Comment:
You can remove this else because the `if` and `elif` both return something, so removing the `else` will help speed up your program a tiny bit.